docs/tutorials/react18-migration.md

```diff
@@ -1,75 +1,127 @@
+       "@types/react": "^18",
+       "@types/react-dom": "^18",
+       // highlight-add-end
+     },
+   ```
+
+2. Update the `react` and `react-dom` dependencies in `packages/app/package.json`:
+
+   ```json title="packages/app/package.json"
+     "dependencies": {
+       ...
+       // highlight-remove-start
+       "react": "^17.0.2",
+       "react-dom": "^17.0.2",
+       // highlight-remove-end
+       // highlight-add-start
+       "react": "^18.0.2",
+       "react-dom": "^18.0.2",
+       // highlight-add-end
+       ...
+     },
+   ```
+
+3. Adjust `packages/app/src/index.tsx` to use the `react-dom/client` API for rendering:
+
+   ```tsx title="packages/app/src/index.tsx"
+   import '@backstage/cli/asset-types';
+   import React from 'react';
+   // highlight-remove-next-line
+   import ReactDOM from 'react-dom';
+   // highlight-add-next-line
+   import ReactDOM from 'react-dom/client';
+   import App from './App';
+
+   // highlight-remove-next-line
+   ReactDOM.render(<App />, document.getElementById('root'));
+   // highlight-add-next-line
+   ReactDOM.createRoot(document.getElementById('root')!).render(<App />);
+   ```
+
+#### Backstage Frontend Plugin
+
+1. Update the `devDependencies` and `peerDependencies` for `react`, `react-dom`, and `@types/react` in your plugin's `package.json`:
+
+   ```json title="plugins/<plugin-name>/package.json"
+     "devDependencies": {
+       ...
+       // highlight-remove-start
+       "@types/react": "^16.13.1 || ^17.0.0",
+       "react": "^16.13.1 || ^17.0.0",
+       "react-dom": "^16.13.1 || ^17.0.0",
+       // highlight-remove-end
+       // highlight-add-start
+       "@types/react": "^17.0.0 || ^18.0.0",
+       "react": "^17.0.0 || ^18.0.0",
+       "react-dom": "^17.0.0 || ^18.0.0",
+       // highlight-add-end
+       ...
+     },
+   ```
+
+   ```json title="plugins/<plugin-name>/package.json"
+     "peerDependencies": {
+       ...
+       // highlight-remove-start
+       "@types/react": "^16.13.1 || ^17.0.0",
+       "react": "^16.13.1 || ^17.0.0",
+       "react-dom": "^16.13.1 || ^17.0.0",
+       // highlight-remove-end
+       // highlight-add-start
+       "@types/react": "^17.0.0 || ^18.0.0",
+       "react": "^17.0.0 || ^18.0.0",
+       "react-dom": "^17.0.0 || ^18.0.0",
+       // highlight-add-end
+       ...
+     },
+   ```
+
+After completing these updates, your application and plugins should function as before, now utilizing React 18.
+
+:::note
+
+Be sure to update your lockfile after modifying your `package.json` files.
+
+:::
 
 ### TypeScript Errors
 
 When upgrading to React 18 you are likely to see a fair number of TypeScript type errors. A summary of the breaking changes can be found in the [Pull Request that introduced them](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210). A codemod is also provided to help with the migration.
 
-Run `yarn tsc:full` to asses the damage.
+Run `yarn tsc:full` to access the damage.
```

Comment:
I think you do actually mean `assess` right? like to make an assessment of the damage?